ush/python/pygw/src/pygw/logger.py

```diff
@@ -0,0 +1,176 @@
+        return formatter.format(record)
+
+
+class Logger:
+    """
+    Improved logging
+    """
+    LOG_LEVELS = ['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL']
+    DEFAULT_LEVEL = 'INFO'
+    DEFAULT_FORMAT = '%(asctime)s - %(levelname)-8s - %(name)-12s: %(message)s'
+    def __init__(self, name: str = None,
+                 level: str = DEFAULT_LEVEL,
+                 _format: str = DEFAULT_FORMAT,
+                 colored_log: bool = False,
+                 logfile_path: Union[str, Path] = None):
+        """
+        Initialize Logger
+
+        Parameters
+        ----------
+        name         : str
+                       Name of the Logger object
+                       default : None
+        level        : str
+                       Desired Logging level
+                       default : 'INFO'
+        _format      : str
+                       Desired Logging Format
+                       default : '%(asctime)s - %(levelname)-8s - %(name)-12s: %(message)s'
+        colored_log  : bool
+                       Use colored logging for stdout
+                       default: False
+        logfile_path : str or Path
+                       Path for logging to a file
+                       default : None
+        """
+
+        self.name = name
+        self.level = level.upper()
+        self.format = _format
+        self.colored_log = colored_log
+
+        if self.level not in Logger.LOG_LEVELS:
+            raise LookupError('{self.level} is unknown logging level\n' +
+                              'Currently supported log levels are:\n' +
+                              f'{" | ".join(Logger.LOG_LEVELS)}')
+
+        # Initialize the root logger if no name is present
+        self._logger = logging.getLogger(name) if name else logging.getLogger()
+
+        self._logger.setLevel(self.level)
+
+        _handlers = []
+        # Add console handler for logger
+        _handler = Logger.add_stream_handler(
+            level=self.level,
+            _format=self.format,
+            colored_log=self.colored_log,
+            )
+        _handlers.append(_handler)
+        self._logger.addHandler(_handler)
+
+        # Add file handler for logger
+        if logfile_path is not None:
+            _handler = Logger.add_file_handler(logfile_path, level=self.level, _format=self.format)
+            self._logger.addHandler(_handler)
+            _handlers.append(_handler)
+
+    def __getattr__(self, attribute):
+        """
+        Allows calling logging module methods directly
+        """
+        return getattr(self._logger, attribute)
+
+    def get_logger(self):
+        """
+        Return the logging object
+
+        Returns
+        -------
+        logger : Logger object
+        """
+        return self._logger
+
+    @classmethod
+    def add_handlers(cls, logger: logging.Logger, handlers: List[logging.Handler]):
+        """
+        Add a list of handlers to a logger
+
+        Parameters
```

Comment:
What should these parameters be? What are their types? (I can get the second by looking at the type hints, but these docstrings will eventually be used by Sphinx or something to create stand-alone documentation webpages.)